dbt/include/spark/macros/adapters.sql

```diff
@@ -134,7 +134,10 @@
 
 {#-- We can't use temporary tables with `create ... as ()` syntax --#}
 {% macro spark__create_temporary_view(relation, compiled_code) -%}
-    create temporary view {{ relation }} as
+    create temporary view {{ relation }}
+    {% if config.get('contract', False) -%}
+      {{ get_assert_columns_equivalent(sql) }}
```

Comment:
Good catch.  I dug in here because of concern for python models (not currently supported) and this code is only hit for incremental models that already exist as tables.  https://github.com/dbt-labs/dbt-core/issues/6755 is the task to make sure those cases are covered so I'm removing this logic as it does not directly relate to models materialized specifically as view.